```

    CC credential.o
config.c: In function 'git_parse_signed':
config.c:873:10: error: absolute value function 'labs' given an argument of type 'intmax_t' {aka 'long long int'} but has parameter of type 'long int' which may cause truncation of value [-Werror=absolute-value]
  873 |   uval = labs(val);
      |          ^~~~
config.c:875:21: error: absolute value function 'labs' given an argument of type 'intmax_t' {aka 'long long int'} but has parameter of type 'long int' which may cause truncation of value [-Werror=absolute-value]
    CC csum-file.o
    CC ctype.o
  875 |   if (uval > max || labs(val) > uval) {
      |                     ^~~~
    CC date.o
cc1.exe: all warnings being treated as errors
    CC delta-islands.o
make: *** [Makefile:2383: config.o] Error 1
make: *** Waiting for unfinished jobs....
    CC decorate.o
```
so many cross-plat things to keep in mind. :(

--- a/cache-tree.c
+++ b/cache-tree.c
@@ -221,7 +221,7 @@ static void discard_unused_subtrees(struct cache_tree *it)
 	}
 }
 
-int cache_tree_fully_valid(struct cache_tree *it)
+static int cache_tree_fully_valid_1(struct cache_tree *it)
 {
 	int i;
 	if (!it)
@@ -229,10 +229,21 @@ int cache_tree_fully_valid(struct cache_tree *it)
 	if (it->entry_count < 0 || !has_object_file(&it->oid))
 		return 0;
 	for (i = 0; i < it->subtree_nr; i++) {
-		if (!cache_tree_fully_valid(it->down[i]->cache_tree))
+		if (!cache_tree_fully_valid_1(it->down[i]->cache_tree))
 			return 0;
 	}
 	return 1;
+}
+
+int cache_tree_fully_valid(struct cache_tree *it)
+{
+	int result;
+
+	trace2_region_enter("cache_tree", "fully_valid", NULL);
+	result = cache_tree_fully_valid_1(it);
+	trace2_region_leave("cache_tree", "fully_valid", NULL);
+
+	return result;
 }
 
 static int update_one(struct cache_tree *it,

--- a/unpack-trees.c
+++ b/unpack-trees.c
@@ -1478,6 +1478,8 @@ int unpack_trees(unsigned len, struct tree_desc *t, struct unpack_trees_options 
 	if (len > MAX_UNPACK_TREES)
 		die("unpack_trees takes at most %d trees", MAX_UNPACK_TREES);
 
+	trace2_region_enter("exp", "unpack_trees", NULL);
+
 	trace_performance_enter();
 	memset(&el, 0, sizeof(el));
 	if (!core_apply_sparse_checkout || !o->update)
@@ -1658,6 +1660,7 @@ int unpack_trees(unsigned len, struct tree_desc *t, struct unpack_trees_options 
 done:
 	trace_performance_leave("unpack_trees");
 	clear_exclude_list(&el);
+	trace2_region_leave("exp", "unpack_trees", NULL);
 	return ret;
 
 return_failed:

--- a/virtualfilesystem.c
+++ b/virtualfilesystem.c
@@ -259,6 +259,8 @@ void apply_virtualfilesystem(struct index_state *istate)
 	if (!git_config_get_virtualfilesystem())
 		return;
 
+	trace2_region_enter("vfs", "apply", the_repository);
+
 	if (!virtual_filesystem_data.len)
 		get_virtual_filesystem_data(&virtual_filesystem_data);
 
@@ -329,6 +331,8 @@ void apply_virtualfilesystem(struct index_state *istate)
 		trace2_data_intmax("vfs", the_repository, "apply/nr_bulk_skip", nr_bulk_skip);
 		trace2_data_intmax("vfs", the_repository, "apply/nr_explicit_skip", nr_explicit_skip);
 	}
+
+	trace2_region_leave("vfs", "apply", the_repository);
 }
 
 /*

